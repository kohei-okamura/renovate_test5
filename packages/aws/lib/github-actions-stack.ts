/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import {
  Stack,
  StackProps,
  aws_iam as iam
} from 'aws-cdk-lib'
import { Construct } from 'constructs'

export type GithubActionsStackProps = StackProps & {
  github: {
    owner: string
    repo: string
  }
  managedPolicies: iam.IManagedPolicy[]
  oidcProviderId: string
  roleId: string
  roleName: string
}

/**
 * GitHub Actions 用 OIDC Provider & IAM Role.
 *
 * See below:
 * - https://dev.classmethod.jp/articles/github-actions-configure-aws-credentials-oidc/
 * - https://qiita.com/takaaki_inada/items/2028328231d1085fa561
 * - https://stackoverflow.com/questions/69247498/how-can-i-calculate-the-thumbprint-of-an-openid-connect-server
 */
export class GithubActionsStack extends Stack {
  constructor (scope: Construct, id: string, props: GithubActionsStackProps) {
    super(scope, id, props)

    const { github, managedPolicies, oidcProviderId, roleId, roleName } = props

    //
    // OIDC Provider.
    //
    const oidcProvider = new iam.OpenIdConnectProvider(this, oidcProviderId, {
      url: 'https://token.actions.githubusercontent.com',
      clientIds: ['sts.amazonaws.com'],
      thumbprints: ['6938fd4d98bab03faadb97b34396831e3780aea1']
    })

    //
    // IAM Role.
    //
    new iam.Role(this, roleId, {
      assumedBy: new iam.FederatedPrincipal(
        oidcProvider.openIdConnectProviderArn,
        {
          StringLike: {
            'token.actions.githubusercontent.com:sub': `repo:${github.owner}/${github.repo}:*`
          }
        },
        'sts:AssumeRoleWithWebIdentity'
      ),
      managedPolicies,
      roleName
    })
  }
}
