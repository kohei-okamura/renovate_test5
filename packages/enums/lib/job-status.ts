/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 *
 * THIS CODE IS AUTO GENERATED. DO NOT EDIT DIRECTLY.
 */
import { createEnumerable } from './enum'

const $$ = {
  waiting: 1,
  inProgress: 2,
  success: 3,
  failure: 9
} as const

/**
 * 非同期ジョブ：状態.
 */
export type JobStatus = typeof $$[keyof typeof $$]

// eslint-disable-next-line @typescript-eslint/no-redeclare -- intentionally naming the variable the same as the type
export const JobStatus = createEnumerable($$, [
  [$$.waiting, '待機中'],
  [$$.inProgress, '処理中'],
  [$$.success, '成功'],
  [$$.failure, '失敗']
])

export const resolveJobStatus = JobStatus.resolve
