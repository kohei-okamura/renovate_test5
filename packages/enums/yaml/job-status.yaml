# Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
# UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.

description: 非同期ジョブ：状態
name: JobStatus
php:
  enabled: true
  namespace: Job
typescript:
  enabled: true
items:
  - { description: 待機中, name: Waiting, value: 1 }
  - { description: 処理中, name: InProgress, value: 2 }
  - { description: 成功, name: Success, value: 3 }
  - { description: 失敗, name: Failure, value: 9 }
