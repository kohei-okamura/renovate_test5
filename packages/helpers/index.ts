/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
export { assert } from './lib/assert'
export { assign } from './lib/assign'
export { camelToKebab } from './lib/camel-to-kebab'
export { camelToSnake } from './lib/camel-to-snake'
export { clone } from './lib/clone'
export { debounce } from './lib/debounce'
export { distinct } from './lib/distinct'
export { identity } from './lib/identity'
export { isFunction } from './lib/is-function'
export { isEmpty } from './lib/is-empty'
export { nonEmpty } from './lib/non-empty'
export { noop } from './lib/noop'
export { keys } from './lib/keys'
export { omit } from './lib/omit'
export { pick } from './lib/pick'
export { range } from './lib/range'
export { tap } from './lib/tap'
export { snakeToCamel } from './lib/snake-to-camel'
export { wait } from './lib/wait'
