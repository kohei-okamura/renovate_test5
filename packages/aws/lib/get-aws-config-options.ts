/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { assert } from '@zinger/helpers'
import { CredentialsOptions } from 'aws-sdk/lib/credentials'
import { execSync } from 'child_process'
import * as fs from 'fs'
import { Seq } from 'immutable'
import ini from 'ini'
import { DateTime } from 'luxon'
import path from 'path'

export type AwsConfigOptions = {
  credentialsOptions: CredentialsOptions
  region: string
}

type AwsCliCache = {
  Credentials: {
    AccessKeyId: string
    SecretAccessKey: string
    SessionToken: string
    Expiration: string
  }
}

type AwsConfigKey = 'region' | 'role_arn' | 'source_profile'
type AwsConfig = Record<AwsConfigKey, string>

const AWS_CONFIG = `${process.env.HOME}/.aws/config`
const AWS_CLI_CACHE = `${process.env.HOME}/.aws/cli/cache`

const normalizeArn = (arn: string): string => arn.replace(':iam:', ':sts:').replace(':role', ':assumed-role')
const loadAwsConfig = (profile: string): AwsConfig => {
  const content = fs.readFileSync(AWS_CONFIG, 'utf-8')
  const config = ini.parse(content)[`profile ${profile}`]
  assert(config !== undefined, `Failed to load config for profile: ${profile}`)
  return config
}

const loadAwsCache = (config: AwsConfig): AwsCliCache | undefined => {
  const cacheFilePaths = fs
    .readdirSync(AWS_CLI_CACHE)
    .filter(filename => filename.toLowerCase().endsWith('.json'))
    .map(filename => path.join(AWS_CLI_CACHE, filename))
  const roleArn = normalizeArn(config.role_arn)
  return Seq(cacheFilePaths)
    .filter(filepath => fs.statSync(filepath).isFile())
    .map(filepath => fs.readFileSync(filepath, 'utf-8'))
    .map(content => JSON.parse(content))
    .find(data => normalizeArn(data.AssumedRoleUser.Arn).includes(roleArn))
}

const isCacheExpired = (cache: AwsCliCache): boolean => {
  return DateTime.fromISO(cache.Credentials.Expiration) <= DateTime.local()
}

const loadAwsCacheGreedy = (profile: string, config: AwsConfig): AwsCliCache | undefined => {
  const cache = loadAwsCache(config)
  if (cache === undefined || isCacheExpired(cache)) {
    execSync(`aws sts get-caller-identity --profile ${profile}`)
    return loadAwsCache(config)
  } else {
    return cache
  }
}

export const getAwsConfigOptions = (profile: string): AwsConfigOptions => {
  const config = loadAwsConfig(profile)
  const cache = loadAwsCacheGreedy(profile, config)

  assert(cache !== undefined, 'Failed to authentication')

  return {
    credentialsOptions: {
      accessKeyId: cache.Credentials.AccessKeyId,
      secretAccessKey: cache.Credentials.SecretAccessKey,
      sessionToken: cache.Credentials.SessionToken
    },
    region: config.region
  }
}
