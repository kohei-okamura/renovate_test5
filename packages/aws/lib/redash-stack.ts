/*
 * Copyright © 2022 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { tap } from '@zinger/helpers'
import {
  Duration,
  Stack,
  StackProps,
  aws_backup as backup,
  aws_certificatemanager as certificateManager,
  aws_ec2 as ec2,
  aws_elasticloadbalancingv2 as elbv2,
  aws_elasticloadbalancingv2_targets as elbv2Targets,
  aws_events as events,
  aws_iam as iam,
  aws_route53 as route53,
  aws_route53_targets as route53Targets
} from 'aws-cdk-lib'
import { Construct } from 'constructs'
import {
  PORT_HTTP, PORT_HTTPS,
  REDASH_ELASTIC_IP,
  REDASH_INSTANCE,
  REDASH_TARGET_GROUP,
  ZINGER_ALB
} from '~aws/variables'

export type RedashStackProps = StackProps & {
  machineImageName: string
  needRedash: boolean
  redashDomainName: string
  vpc: ec2.Vpc
  zoneDomainName: string
}

export class RedashStack extends Stack {
  constructor (scope: Construct, id: string, props: RedashStackProps) {
    super(scope, id, props)

    const { machineImageName, needRedash, redashDomainName, vpc, zoneDomainName } = props

    if (!needRedash) { return }

    //
    // Redash用 EC2 ROLE 定義
    //
    const role = (id: string, roleProps: iam.RoleProps): iam.IRole => {
      return new iam.Role(this, id, {
        ...roleProps
      })
    }
    const redashRole = role('RedashIamRole', {
      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonS3FullAccess'),
        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore')
      ],
      roleName: 'ZingerRedashRole'
    })

    //
    // Redash用 securityGroup 定義
    //
    const securityGroup = tap(
      new ec2.SecurityGroup(this, 'RedashSecurityGroup', { vpc }),
      it => {
        it.addIngressRule(ec2.Peer.anyIpv4(), ec2.Port.tcp(PORT_HTTP))
      }
    )

    //
    // Redash用 EC2 定義
    //
    const redashInstance = new ec2.Instance(this, REDASH_INSTANCE, {
      blockDevices: [{
        deviceName: '/dev/sda1',
        volume: ec2.BlockDeviceVolume.ebs(20)
      }],
      instanceName: REDASH_INSTANCE,
      instanceType: new ec2.InstanceType('t3a.small'),
      machineImage: ec2.MachineImage.lookup({
        name: machineImageName
      }),
      role: redashRole,
      securityGroup,
      userDataCausesReplacement: false,
      vpc,
      vpcSubnets: { subnetGroupName: 'Public' }
    })

    //
    // Redash用 EC2 Elastic IP 定義
    //
    const eip = new ec2.CfnEIP(this, REDASH_ELASTIC_IP, {
      tags: [{ key: 'Name', value: REDASH_ELASTIC_IP }]
    })
    new ec2.CfnEIPAssociation(this, 'RedashElasticIpAssociation', {
      eip: eip.ref,
      instanceId: redashInstance.instanceId
    })

    //
    // Redash用 EC2 AWS Backup 定義
    //
    const plan = new backup.BackupPlan(this, 'RedashBackupPlan', {
      backupPlanName: 'RedashBackupPlan'
    })
    plan.addRule(new backup.BackupPlanRule({
      completionWindow: Duration.hours(2),
      startWindow: Duration.hours(1),
      scheduleExpression: events.Schedule.cron({
        hour: '17',
        minute: '35'
      }),
      deleteAfter: Duration.days(7)
    }))
    new backup.BackupSelection(this, 'RedashBackupSelection', {
      backupPlan: plan,
      resources: [
        backup.BackupResource.fromEc2Instance(redashInstance)
      ]
    })

    //
    // Redash用 Route 53 A/AAAA レコード定義
    //
    // 本来ならば ApplicationLoadBalancer のオブジェクトは、ZingerAlbDnsStack から参照することが望ましいが、
    // cdkを実行するとARN（実体）を取得できない参照エラーとなる為、作成済みのリソースから情報を取得する `〜.fromlookup`
    // からオブジェクトを取得する。
    const alb = elbv2.ApplicationLoadBalancer.fromLookup(this, 'RedashAlb', {
      loadBalancerTags: { Name: ZINGER_ALB }
    })
    const hostedZone = route53.HostedZone.fromLookup(this, 'RedashHostedZone', {
      domainName: zoneDomainName
    })
    const redashRoute53Record = {
      recordName: redashDomainName,
      target: route53.RecordTarget.fromAlias(
        new route53Targets.LoadBalancerTarget(alb)
      ),
      zone: hostedZone
    }
    new route53.ARecord(this, 'RedashARecord', redashRoute53Record)
    new route53.AaaaRecord(this, 'RedashAaaaRecord', redashRoute53Record)

    //
    // Redash用 ALB Target Group 定義
    //
    const redashTargetGroup = new elbv2.ApplicationTargetGroup(this, REDASH_TARGET_GROUP, {
      port: PORT_HTTP,
      targetGroupName: REDASH_TARGET_GROUP,
      targetType: elbv2.TargetType.INSTANCE,
      targets: [new elbv2Targets.InstanceTarget(redashInstance)],
      vpc
    })

    //
    // Redash用 ALBへ関連付けるSSL証明書 定義
    //
    const redashCertificate = new certificateManager.DnsValidatedCertificate(this, 'RedashCertificate', {
      domainName: redashDomainName,
      hostedZone,
      region: vpc.env.region
    })

    //
    // Redash用 Route53 & ALB
    //
    const listener = elbv2.ApplicationListener.fromLookup(this, 'RedashAlbListener', {
      loadBalancerArn: alb.loadBalancerArn,
      listenerPort: PORT_HTTPS,
      listenerProtocol: elbv2.ApplicationProtocol.HTTPS
    })
    listener.addCertificates('RedashAddCertificates', [redashCertificate])

    //
    // Redash用 ALB HTTP 接続定義 を追加
    //
    new elbv2.ApplicationListenerRule(this, 'RedashListenerRule', {
      conditions: [
        elbv2.ListenerCondition.hostHeaders([redashDomainName])
      ],
      listener,
      priority: 1,
      targetGroups: [redashTargetGroup]
    })
  }
}
