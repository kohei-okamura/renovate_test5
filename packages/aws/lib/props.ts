/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import {
  StackProps,
  aws_ec2 as ec2,
  aws_iam as iam,
  aws_s3 as s3,
  aws_ssm as ssm
} from 'aws-cdk-lib'
import { GithubActionsStackProps } from '~aws/lib/github-actions-stack'
import { RedashStackProps } from '~aws/lib/redash-stack'

export type ZingerAlbDnsStackProps = StackProps & {
  logStorage: s3.Bucket
  public: {
    aRecord: string
    domainName: string
  }
  securityGroup: ec2.SecurityGroup
  vpc: ec2.Vpc
}

export type ZingerBastionStackProps = StackProps & {
  bastionRole: iam.IRole
  bastionHostname: string
  keyName: string
  securityGroup: ec2.SecurityGroup
  subnetGroupName: string
  vpc: ec2.Vpc
}

export type ZingerDbStackProps = StackProps & {
  instanceType: ec2.InstanceType
  masterUserPassword: ssm.SecureStringParameterAttributes
  multiAz: boolean
  securityGroup: ec2.SecurityGroup
  vpc: ec2.Vpc
}

type Repository = {
  id: string
  repositoryName: string
}

export type ZingerEcrStackProps = StackProps & {
  repositories: Repository[]
}

export type ZingerEcsStackProps = StackProps & {
  name: string
  vpc: ec2.Vpc
}

export type ZingerRedisStackProps = StackProps & {
  authToken: ssm.SecureStringParameterAttributes
  automaticFailoverEnabled: boolean
  cacheNodeType: string
  multiAzEnabled: boolean
  numCacheClusters: number
  securityGroup: ec2.SecurityGroup
  subnetGroupName: string
  vpc: ec2.Vpc
}

export type ZingerS3StackProps = StackProps & {
  suffix?: string
}

export type ZingerSecurityGroupStackProps = StackProps & {
  vpc: ec2.Vpc
}

export type ZingerVpcStackProps = StackProps & {
  cidr: string
  logStorage: s3.Bucket
  natGateways: number
  onePerAz: boolean
}

export type ZingerProps = {
  account: string
  albDns: Omit<ZingerAlbDnsStackProps, 'logStorage' | 'securityGroup' | 'vpc'>
  bastion: Omit<ZingerBastionStackProps, 'bastionRole' | 'securityGroup' | 'subnetGroupName' | 'vpc'>
  db: Omit<ZingerDbStackProps, 'securityGroup' | 'vpc'>
  ecr: ZingerEcrStackProps
  ecs: Omit<ZingerEcsStackProps, 'vpc'>
  githubActions: GithubActionsStackProps
  redash: Omit<RedashStackProps, 'vpc'>
  redis: Omit<ZingerRedisStackProps, 'securityGroup' | 'subnetGroupName' | 'vpc'>
  s3: ZingerS3StackProps
  vpc: Omit<ZingerVpcStackProps, 'logStorage'>
}
