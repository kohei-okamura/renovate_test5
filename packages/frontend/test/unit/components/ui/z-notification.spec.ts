/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { JobStatus } from '@zinger/enums/lib/job-status'
import ZNotification from '~/components/ui/z-notification.vue'
import { createMockedRouter } from '~~/test/helpers/create-mocked-router'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

describe('z-notification.vue', () => {
  const { mount } = setupComponentTest()

  it.each([
    ['waiting', JobStatus.waiting],
    ['inProgress', JobStatus.inProgress],
    ['success', JobStatus.success],
    ['failure', JobStatus.failure]
  ])('should be rendered correctly when status is "%s"', (_, status) => {
    const propsData = {
      id: 1,
      status,
      linkToOnFailure: 'home',
      linkToOnSuccess: 'dashboard'
    }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper).toMatchSnapshot()
  })

  it('should be rendered conditional elements correctly when status is "inProgress"', () => {
    const propsData = {
      id: 1,
      status: JobStatus.inProgress,
      linkToOnFailure: 'home',
      linkToOnSuccess: 'dashboard'
    }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper.find('[data-in-progress-icon]')).toExist()
    expect(wrapper.find('[data-success-icon]')).not.toExist()
    expect(wrapper.find('[data-failure-icon]')).not.toExist()
    expect(wrapper.find('[data-close-button]')).not.toExist()
    expect(wrapper.find('[data-detail-button-container-on-success]')).not.toExist()
    expect(wrapper.find('[data-detail-button-container-on-failure]')).not.toExist()
  })

  it('should be rendered conditional elements correctly when status is "success" (does not have linkToOnSuccess)', () => {
    const propsData = {
      id: 1,
      status: JobStatus.success,
      linkToOnFailure: 'home'
    }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper.find('[data-in-progress-icon]')).not.toExist()
    expect(wrapper.find('[data-success-icon]')).toExist()
    expect(wrapper.find('[data-failure-icon]')).not.toExist()
    expect(wrapper.find('[data-close-button]')).toExist()
    expect(wrapper.find('[data-detail-button-container-on-success]')).not.toExist()
    expect(wrapper.find('[data-detail-button-container-on-failure]')).not.toExist()
  })

  it('should be rendered conditional elements correctly when status is "success" (has linkToOnSuccess)', () => {
    const propsData = {
      id: 1,
      status: JobStatus.success,
      linkToOnFailure: 'home',
      linkToOnSuccess: 'dashboard'
    }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper.find('[data-in-progress-icon]')).not.toExist()
    expect(wrapper.find('[data-success-icon]')).toExist()
    expect(wrapper.find('[data-failure-icon]')).not.toExist()
    expect(wrapper.find('[data-close-button]')).toExist()
    expect(wrapper.find('[data-detail-button-container-on-success]')).toExist()
    expect(wrapper.find('[data-detail-button-container-on-failure]')).not.toExist()
  })

  it('should be rendered conditional elements correctly when status is "failure" (does not have linkToOnFailure)', () => {
    const propsData = {
      id: 1,
      status: JobStatus.failure,
      linkToOnSuccess: 'dashboard'
    }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper.find('[data-in-progress-icon]')).not.toExist()
    expect(wrapper.find('[data-success-icon]')).not.toExist()
    expect(wrapper.find('[data-failure-icon]')).toExist()
    expect(wrapper.find('[data-close-button]')).toExist()
    expect(wrapper.find('[data-detail-button-container-on-success]')).not.toExist()
    expect(wrapper.find('[data-detail-button-container-on-failure]')).not.toExist()
  })

  it('should be rendered conditional elements correctly when status is "failure" (has linkToOnFailure)', () => {
    const propsData = {
      id: 1,
      status: JobStatus.failure,
      linkToOnFailure: 'home',
      linkToOnSuccess: 'dashboard'
    }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper.find('[data-in-progress-icon]')).not.toExist()
    expect(wrapper.find('[data-success-icon]')).not.toExist()
    expect(wrapper.find('[data-failure-icon]')).toExist()
    expect(wrapper.find('[data-close-button]')).toExist()
    expect(wrapper.find('[data-detail-button-container-on-success]')).not.toExist()
    expect(wrapper.find('[data-detail-button-container-on-failure]')).toExist()
  })

  it.each([
    ['inProgress', JobStatus.inProgress],
    ['success', JobStatus.success],
    ['failure', JobStatus.failure]
  ])('should be rendered featureName if props.featureName is passed [status is "%s"]', (_, status) => {
    const featureName = 'My Awesome Feature'
    const propsData = { id: 1, status, featureName }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper.text()).toContain(featureName)
  })

  it.each([
    ['inProgress', [JobStatus.inProgress, 'in progress']],
    ['success', [JobStatus.success, 'terminating successfully']],
    ['failure', [JobStatus.failure, 'failing and ending']]
  ])('should be rendered text if props.text is passed [status is "%s"]', (_, [status, str]) => {
    const text = `The process is currently ${str}!`
    const propsData = { id: 1, status, text }
    const wrapper = mount(ZNotification, { propsData })
    expect(wrapper.text()).toContain(text)
  })

  it('should emit "click" with id when close button clicked', async () => {
    const mockFn = jest.fn()
    const id = 12345
    const wrapper = mount({
      data: () => {
        return {
          id,
          status: JobStatus.success,
          handleClick: mockFn
        }
      },
      template: '<z-notification :id="id" :status="status" @click="handleClick" />',
      components: { 'z-notification': ZNotification }
    })
    const buttonWrapper = wrapper.find('[data-close-button]')
    await buttonWrapper.trigger('click')
    expect(mockFn).toHaveBeenCalledTimes(1)
    expect(mockFn).toHaveBeenCalledWith(id)
  })

  it('should emit "click" with id when detail button clicked on failure', async () => {
    const $router = createMockedRouter()
    const spy = jest.spyOn($router, 'push')
    const mockFn = jest.fn()
    const id = 54321
    const linkToOnFailure = 'home'
    const wrapper = mount({
      data: () => {
        return {
          id,
          status: JobStatus.failure,
          linkToOnFailure,
          handleClick: mockFn
        }
      },
      template: '<z-notification :id="id" :status="status" :link-to-on-failure="linkToOnFailure" @click="handleClick" />',
      components: { 'z-notification': ZNotification }
    }, { mocks: { $router } })
    const buttonWrapper = wrapper.find('[data-detail-button-on-failure]')
    await buttonWrapper.trigger('click')
    expect(mockFn).toHaveBeenCalledTimes(1)
    expect(mockFn).toHaveBeenCalledWith(id)
    expect(spy).toHaveBeenCalledTimes(1)
    expect(spy).toHaveBeenCalledWith(linkToOnFailure)
    spy.mockRestore()
  })

  it('should emit "click" with id when detail button clicked on success', async () => {
    const $router = createMockedRouter()
    const spy = jest.spyOn($router, 'push')
    const mockFn = jest.fn()
    const id = 54321
    const linkToOnSuccess = 'home'
    const wrapper = mount({
      data: () => {
        return {
          id,
          status: JobStatus.success,
          linkToOnSuccess,
          handleClick: mockFn
        }
      },
      template: '<z-notification :id="id" :status="status" :link-to-on-success="linkToOnSuccess" @click="handleClick" />',
      components: { 'z-notification': ZNotification }
    }, { mocks: { $router } })
    const buttonWrapper = wrapper.find('[data-detail-button-on-success]')
    await buttonWrapper.trigger('click')
    expect(mockFn).toHaveBeenCalledTimes(1)
    expect(mockFn).toHaveBeenCalledWith(id)
    expect(spy).toHaveBeenCalledTimes(1)
    expect(spy).toHaveBeenCalledWith(linkToOnSuccess)
    spy.mockRestore()
  })
})
