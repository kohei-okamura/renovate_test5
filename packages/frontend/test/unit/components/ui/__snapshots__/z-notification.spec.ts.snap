// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-notification.vue should be rendered correctly when status is "failure" 1`] = `
<div
  class="px-3 py-2 root"
  data-z-notification=""
>
  <div
    class="d-inline-flex"
  >
    <div
      class="mr-3"
    >
      <v-icon-stub
        color="error"
        data-failure-icon=""
        datafailureicon=""
      >
        M12,2L1,21H23M12,6L19.53,19H4.47M11,10V14H13V10M11,16V18H13V16
      </v-icon-stub>
    </div>
    <div>
      <span
        class="font-weight-bold"
      >
        完了
      </span>
      <span>
        -
      </span>
      <span
        class="body-1"
      >
        タスクの処理にエラーがありました
      </span>
    </div>
    <a
      class="v-btn-stub"
      color="transparent"
      data-close-button=""
      height="24"
      icon=""
      small=""
    >
      <v-icon-stub
        color="primary"
      >
        M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z
      </v-icon-stub>
    </a>
  </div>
  <div
    class="d-inline-flex justify-end container"
    data-detail-button-container-on-failure=""
  >
    <a
      class="v-btn-stub"
      color="primary"
      data-detail-button-on-failure=""
      text=""
    >
      詳細を確認
    </a>
  </div>
</div>
`;

exports[`z-notification.vue should be rendered correctly when status is "inProgress" 1`] = `
<div
  class="px-3 py-2 root"
  data-z-notification=""
>
  <div
    class="d-inline-flex"
  >
    <div
      class="mr-3"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="v-progress-circular v-progress-circular--visible v-progress-circular--indeterminate blue-grey--text"
        data-in-progress-icon=""
        role="progressbar"
        style="height: 26px; width: 26px;"
      >
        <svg
          style="transform: rotate(0deg);"
          viewBox="22.608695652173914 22.608695652173914 45.21739130434783 45.21739130434783"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            class="v-progress-circular__overlay"
            cx="45.21739130434783"
            cy="45.21739130434783"
            fill="transparent"
            r="20"
            stroke-dasharray="125.664"
            stroke-dashoffset="125.66370614359172px"
            stroke-width="5.217391304347826"
          />
        </svg>
        <div
          class="v-progress-circular__info"
        />
      </div>
    </div>
    <div>
      <span
        class="font-weight-bold"
      >
        処理中
      </span>
      <span>
        -
      </span>
      <span
        class="body-1"
      >
        タスクを処理しています
      </span>
    </div>
    <!---->
  </div>
  <!---->
</div>
`;

exports[`z-notification.vue should be rendered correctly when status is "success" 1`] = `
<div
  class="px-3 py-2 root"
  data-z-notification=""
>
  <div
    class="d-inline-flex"
  >
    <div
      class="mr-3"
    >
      <v-icon-stub
        color="success"
        data-success-icon=""
        datasuccessicon=""
      >
        M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z
      </v-icon-stub>
    </div>
    <div>
      <span
        class="font-weight-bold"
      >
        完了
      </span>
      <span>
        -
      </span>
      <span
        class="body-1"
      >
        タスクの処理が正常に終了しました
      </span>
    </div>
    <a
      class="v-btn-stub"
      color="transparent"
      data-close-button=""
      height="24"
      icon=""
      small=""
    >
      <v-icon-stub
        color="primary"
      >
        M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z
      </v-icon-stub>
    </a>
  </div>
  <div
    class="d-inline-flex justify-end container"
    data-detail-button-container-on-success=""
  >
    <a
      class="v-btn-stub"
      color="primary"
      data-detail-button-on-success=""
      text=""
    >
      詳細を確認
    </a>
  </div>
</div>
`;

exports[`z-notification.vue should be rendered correctly when status is "waiting" 1`] = `
<div
  class="px-3 py-2 root"
  data-z-notification=""
>
  <div
    class="d-inline-flex"
  >
    <div
      class="mr-3"
    >
      <!---->
    </div>
    <div>
      <span
        class="font-weight-bold"
      >
        待機中
      </span>
      <span>
        -
      </span>
      <span
        class="body-1"
      >
        待機中
      </span>
    </div>
    <!---->
  </div>
  <!---->
</div>
`;
